src/u_init.c

```diff
@@ -1093,8 +1093,12 @@ ini_inv(struct trobj *trop)
             free_oname(obj);
         }
 
-        /* Don't have erosion or erosionproofing */
-        obj->oeroded = obj->oeroded2 = obj->oerodeproof = 0;
+        /* Don't have erosion or erosionproofing, except rustproof "lacquered
+         * armor" for samurai (see comment in mksobj) */
+        obj->oeroded = obj->oeroded2 = 0;
+        if (!(Role_if(PM_SAMURAI) && obj->otyp == SPLINT_MAIL)) {
+            obj->oerodeproof = 0;
+        }
```

Comment:
Is this actually necessary any more, now that the "make some items fixed or eroded" bit in `mksobj` depends on `g.in_mklev` being true?

Reply:
I added 435cfac  as a different approach to addressing this... I don't think both 435cfac and fefd024 are needed, but I'm not certain which one is closer to the "right" way to do it.  I'm not going to remove either one from the PR for now, so that you can see them both.